lib/mixlib/install.rb

```diff
@@ -59,5 +61,32 @@ def root
       # we are supporting as of now.
       "/opt/#{options.product_name}"
     end
+
+    #
+    # Returns the current version of the installed product.
+    # Returns nil if the product is not installed.
+    #
+    def current_version
+      # Note that this logic does not work for products other than
+      # chef & chefdk since version-manifest is created under the
+      # install directory which can be different than the product name (e.g.
+      # chef-server -> /opt/opscode). But this is OK for now since
+      # chef & chefdk are the only supported products for now.
+      version_manifest_file = "/opt/#{options.product_name}/version-manifest.json"
+      if File.exist? version_manifest_file
+        JSON.parse(File.read(version_manifest_file))["build_version"]
+      end
+    end
+
+    #
+    # Returns true if an upgradable version is available, false otherwise.
+    #
+    def upgrade_available?
+      return false if current_version.nil?
```

Comment:
Wouldn't we just assume an upgrade is available if the current version can't be determined?